distributed/tests/test_executor.py

```diff
@@ -3784,3 +3786,34 @@ def test_stress_scatter_death(e, s, *workers):
         import pdb; pdb.set_trace()
     except CancelledError:
         pass
+
+
+@gen_cluster(executor=False)
+def test_serialize_future(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
+    f = Executor((s.ip, s.port), start=False)
+    yield f._start()
+
+    future = e.submit(lambda: 1)
+
+    with temp_default_executor(f):
+        future2 = pickle.loads(pickle.dumps(future))
+        assert future2.executor is f
+        assert tokey(future2.key) in f.futures
```

Comment:
'cancelled' is the case when `f.key not in e.futures` .  On deserialization of the `Future` we should add an entry to `e.futures` with an empty event and status.  This normally happens in `Future.__init__`

        if tkey not in executor.futures:
            executor.futures[tkey] = {'event': Event(), 'status': 'pending'}